openssl/src/pkcs7.rs

```diff
@@ -281,11 +296,37 @@ impl Pkcs7Ref {
             Ok(stack)
         }
     }
+
+    // Return the type of a PKCS#7 structure as an Asn1Object
+    pub fn type_oid(&self) -> Option<&Asn1ObjectRef> {
+        unsafe {
+            let ptr = (*self.as_ptr()).type_;
+            Asn1ObjectRef::from_const_ptr_opt(ptr)
+        }
+    }
+
+    // Retrieve all the certificates from a PKCS#7 structure used for signed data
+    pub fn signed_data_certificates(&self) -> Result<Option<&StackRef<X509>>, ErrorStack> {
```

Comment:
I think I'd rather go "1 step" here - instead having a `pub fn signed(&self) -> Option<&Pkcs7SignedRef>` method.

Reply:
Wouldn't this mean that users of this library would have to use unsafe code to extract the stack of certificates from the `Pkcs7SignedRef`? Since they would have to do the extra step that this function now does:
```rust
                .and_then(|x| x.cert.as_mut())
                .and_then(|x| StackRef::<X509>::from_const_ptr_opt(x));
```
which is unsafe.

Reply:
I assume @sfackler was suggesting that `Pkcs7SignedRef` should have safe methods for accessing this data.

Reply:
Got it. Fixed, now the stack of certificates can be accessed in two steps:
```rust
let signed_certificates = pkcs7.signed().and_then(|x| x.certificates());
```